[Woo POS][Refunds] Add Issue Refund button

samiuelson opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
WOOMOB-1801

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR adds an "Issue Refund" button to the WooCommerce POS orders screen with a placeholder dialog implementation for the refund flow.

💡 The `WooPosIssueRefundDialog` layout is temporary. This PR aims to just open a dialog; its layout will be implemented in next PR.

💡 Feel free to remove `do not merge` label once the parent branch becomes `trunk`.

### Test Steps
<!-- Describe how to test your changes. Include only what’s needed for the reviewer to validate the behavior.
For new features, outline the main user flow or goal rather than every tap or click — the reviewer should be able to complete the flow naturally.
If applicable, mention key devices, scenarios, or edge cases to verify. -->

1. Verify that the `Issue refund` button is not showing up when the `POS_REFUNDS` is `false`.
2. Verify that `Email Receipt` button works
3. Verify that `Issue Refund` button opens a dialog with a temporary layout.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
<img width="900"  alt="Screenshot_20251126_184347" src="https://github.com/user-attachments/assets/ea5e0b0e-1425-4dce-bd10-9c0c18e191be" />
<img width="900"  alt="Screenshot_20251126_184354" src="https://github.com/user-attachments/assets/10152294-2b8e-4524-bd01-d8dc5a4a59f0" />


- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Add issue refund dialog and move email receipt to overflow menu

- Update `WooPosOrdersState` to handle dialog state for issuing refunds.
- Implement `WooPosIssueRefundDialog` as a placeholder for the refund flow.
- Update `WooPosOrdersDetails` to replace the "Email receipt" button with an "Issue refund" button.
- Move the "Email receipt" action into a new overflow menu in the order details view.
- Connect the refund button and dialog logic in `WooPosOrdersScreen` and `WooPosOrdersViewModel`.

samiuelson committed:
Add unit tests for Issue Refund dialog in WooPosOrdersViewModel

samiuelson committed:
Satisfy detekt's complaints

samiuelson committed:
Merge branch 'pos-orders-fix-status-bar-handling' into woomob-1801-add-the-issue-refund-button

kidinov reviewed (commented):
👋 @samiuelson 

* The button is visible even when FF is off - so not finished functionality will be released?
* I am not sure about the design of the button—in Figma there is no more overflow menu without an email button at all 🤔 

<img width="538" height="379" alt="Image" src="https://github.com/user-attachments/assets/58ed6e0d-1b51-405e-9651-333ae83dadce" />

samiuelson committed:
Hide refund UI elements when POS_REFUNDS flag is disabled

- Propagate the `POS_REFUNDS` feature flag status through `WooPosOrdersState` and `WooPosOrdersViewModel`.
- Update `WooPosOrdersDetails` to conditionally render the "Issue Refund" button and overflow menu.
- Show a direct "Email Receipt" button when refunds are disabled.

dangermattic commented:
<!--
  1 Error: This PR is tagged with <code>s...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>status: do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosOrdersViewModel.kt:
```diff
@@ -46,8 +47,13 @@ class WooPosOrdersViewModel @Inject constructor(
     private val ordersAnalyticsTracker: WooPosOrdersAnalyticsTracker
 ) : ViewModel() {
 
+    private val isRefundsEnabled = FeatureFlag.POS_REFUNDS.isEnabled()
```
The refund button probably should be hidden also when selected order is fully refunded, right?

samiuelson replied:
Yes, good point. I plan to implement that later: WOOMOB-1773

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosOrdersDetails.kt:
```diff
@@ -69,10 +80,24 @@ fun WooPosOrderDetails(
-                text = stringResource(R.string.woopos_orders_email_receipt),
-                onClick = { onEmailReceiptButtonClicked(details.id) },
-            )
+            if (isRefundsEnabled) {
```
I think it's not a good state model. It looks like we need to expose a list of buttons to the view, where the first one will be visible while the rest will be hidden in the overflow menu. Currently we kinda hardcode the priority in the view, while it's highly likely it will depend on the state of the specific order

kidinov replied:
With the list of buttons, we can pass events that each specific button emits when clicked, just like we do on the home page. Eventually, we’ll have numerous actions that can be performed on the orders.

kidinov reviewed (approved):
I left a couple of comments but they can be implemented as a seprate PR

Also:
<img width="225" height="132" alt="Image" src="https://github.com/user-attachments/assets/544a9cd1-f24d-4eca-ad7c-282fb35f6601" />

The floating menu button doesn't match the design

samiuelson committed:
Refactor POS order actions to use a generic OrderAction list

- Replace `isRefundsEnabled` boolean in `WooPosOrdersState` with a list of `OrderAction` objects.
- Introduce `OrderAction` sealed interface with `IssueRefund` and `EmailReceipt` types.
- Update `WooPosOrdersViewModel` to determine available actions based on feature flags.
- Adjust `WooPosOrdersDetails` UI to render the first action as a button and remaining actions in the overflow menu.

samiuelson committed:
Update DropdownMenu background color in WooPosOrdersDetails

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosOrdersDetails.kt:
```diff
@@ -43,7 +54,9 @@ import com.woocommerce.android.ui.woopos.common.composeui.designsystem.WooPosTyp
     details: OrderDetailsViewState.Computed.Details,
-    onEmailReceiptButtonClicked: (Long) -> Unit
+    actions: List<OrderAction>,
+    onEmailReceiptButtonClicked: (Long) -> Unit,
```
Not sure if it's the final state or still WIP, but this is not something that is driven by the VM as you have to pass hardcoded click listeners here and everywhere instead of calling one `onUIEvent` that accepts events that are part of the “actions”. Also, why is `OrderAction` not part of the whole state but a separate class?

samiuelson replied:
Good idea, done here: d6443ab

kidinov replied:
Looks nice, thanks!

samiuelson committed:
Move order actions to OrderDetailsViewState and introduce WooPosOrdersUIEvent

- Nest `OrderAction`, `OrderDetailsViewState`, and `OrderItemViewState` classes within `WooPosOrdersState`.
- Move the `actions` list from `WooPosOrdersState` subclasses into `OrderDetailsViewState.Computed.Details`.
- Create `WooPosOrdersUIEvent` sealed interface to handle UI events.
- Replace specific callbacks `onEmailReceiptButtonClicked` and `onIssueRefundButtonClicked` with `onUIEvent` in `WooPosOrdersScreen` and `WooPosOrdersDetails`.
- Update `WooPosOrdersViewModel` to compute actions during details mapping and handle `OrderActionClicked` events.

samiuelson committed:
Merge branch 'pos-orders-fix-status-bar-handling' into woomob-1801-add-the-issue-refund-button

samiuelson committed:
Merge branch 'pos-orders-fix-status-bar-handling' into woomob-1801-add-the-issue-refund-button

samiuelson committed:
Merge branch 'trunk' into woomob-1801-add-the-issue-refund-button

samiuelson committed:
Merge branch 'trunk' into woomob-1801-add-the-issue-refund-button

samiuelson committed:
Add missing import
